hawkbit-repository/hawkbit-repository-api/src/main/java/org/eclipse/hawkbit/repository/TargetQueryExecutionManagement.java

```diff
@@ -0,0 +1,107 @@
+/**
+ * Copyright (c) 2019 Bosch Software Innovations GmbH and others.
+ *
+ * All rights reserved. This program and the accompanying materials
+ * are made available under the terms of the Eclipse Public License v1.0
+ * which accompanies this distribution, and is available at
+ * http://www.eclipse.org/legal/epl-v10.html
+ */
+package org.eclipse.hawkbit.repository;
+
+import java.util.Collection;
+
+import javax.validation.constraints.NotEmpty;
+import javax.validation.constraints.NotNull;
+
+import org.eclipse.hawkbit.im.authentication.SpPermission;
+import org.eclipse.hawkbit.im.authentication.SpPermission.SpringEvalExpressions;
+import org.eclipse.hawkbit.repository.model.Target;
+import org.eclipse.hawkbit.repository.model.TargetFilterQuery;
+import org.springframework.data.domain.Page;
+import org.springframework.data.domain.Pageable;
+import org.springframework.security.access.prepost.PreAuthorize;
+
+/**
+ * This interface is the central point which is responsible to execute queries
+ * and return update-targets for Hawkbit.
+ *
+ * By design it should be as minimal as possible in order to make it easy to
```

Comment:
I want to prevent that someone is adding something like a findByFilterAnd<some rollouts-domain-logic-here>. So documenting the intend of an interface is not desired?

Reply:
> I want to prevent that someone is adding something like a `findByFilterAnd<some rollouts-domain-logic-here>`

Got it! What about making this intention more explicit like:

By design it only contains generic filter methods so given query parameters can be used in a more universal, model independent manner. Thus domain specific querying e.g. `findByFilterAndDistributionSet(...)` should rather be expressed within the query syntax than by introducing (new) named methods.

Reply:
:thumbsup: